### PR TITLE
Use the object ID to generate the preview

### LINKED ISSUE #340: Forward port #336 handling of previews to master before 8.2 release
Unless the issue has been fixed in an other way:
https://github.com/owncloud/activity/pull/336

### LINKED ISSUE #369: Append etag in preview URLs
I got a case where the URL returned by the activity app was showing the wrong cached preview from an older version of the file.

Ideal would be to either append a random string "c=1a2b3c4d5" in the URL or the etag.

### LINKED ISSUE #381: Blurry icons with new icons of core
bildschirmfoto von 2015-08-29 13-47-20

core master

cc @jancborchardt

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Tests needs adjustments

Files in trashbin still don't work: https://github.com/owncloud/core/issues/18509
Use the ID in the parameters as well?

Fix #369 
Fix #340 
Fix #381

@PVince81

### CODE REVIEW

**Comment by nickvergessen** [comment]
Fix #369

@PVince81

File: controller/activities.php [hunk 1]
```diff
@@ -178,42 +178,42 @@ public function fetch($page, $filter = 'all', $objecttype = '', $objectid = 0) {
...
 	/**
 	 * @param array $activity
-	 * @param string $file
+	 * @param int $fileId
+	 * @param string $filePath
 	 * @return array
 	 */
-	protected function getPreview(array $activity, $file) {
+	protected function getPreview(array $activity, $fileId, $filePath) {
 		$this->view->chroot('/' . $activity['affecteduser'] . '/files');
```
Comment (nickvergessen) [inline controller/activities.php]: Needs trailing slash and special code for trashbin

**Comment by nickvergessen** [comment]
@MorrisJobke fixes the blurry icon, if you want to test it ;)

**Comment by MorrisJobke** [comment]
Works :+1:

**Pull request merged**